### PR TITLE
chore: [IA-157] Share unique ID with assistance

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
We want to let the users share their unique ID identifier, should they want to. I had to refactor a bit and decided to improve the test coverage in this area and remove what seems redundant logic in BaseScreen.

See [this report](https://dashboard.instabug.com/applications/io-l-app-dei-servizi-pubblici/beta/bugs/2290561?status=Closed) to check for the new attribute:

Screenshot 2021-08-03 at 15 15 11

## List of changes proposed in this pull request
- Update the CTA text [en](locales/en/index.yml) [it](locales/it/index.yml) to include device's UUID (sorry, couldn't handle this one in 🇩🇪 🙇 )
- Group ContextualHelp items in [ts/components/ContextualHelp](ts/components/ContextualHelp) and refactor accordingly
- add one test to check the request assistance button in [ContextualHelpComponent](ts/components/ContextualHelp/__tests__/ContextualHelpComponent.test.tsx)
- add a first (humble) smoke test for [ContextualHelpModal](ts/components/ContextualHelp/__tests__/ContextualHelpModal.test.tsx)
- extract the side-effects from BaseScreen to [a separate function](https://github.com/pagopa/io-app/pull/3266/files#diff-99b6f65a74838b4555bfed0547c21e3f554b6fbca42f36eb9687795a8c2e96b6R46) which can now be [tested](ts/components/screens/__tests__/BaseScreenComponent.test.tsx) 🎉 

## How to test

Find the walkthrough below, nothing has changed besides the checkbox's label.

https://user-images.githubusercontent.com/2094604/128038610-c7e11ec6-49c6-408c-b492-be78b53c07ba.mov

### CODE REVIEW

File: ts/components/ContextualHelp/CheckboxFormEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          onPress={onToggle}
+        >
+          {checkboxLabelKeys[target].cta}
+        </Label>
+        {"accordion" in checkboxLabelKeys[target] && (
+          <Accordion
+            title={checkboxLabelKeys[target].accordion?.title || ""}
+            content={checkboxLabelKeys[target].accordion?.content || ""}
+          />
+        )}
```
Comment (Undermaken) [inline ts/components/ContextualHelp/CheckboxFormEntry.tsx:85]:
```suggestion
        const accordion = checkboxLabelKeys[target].accordion;
        [...]
        {accordion && (
          <Accordion title={accordion.title} content={accordion.content} />
        )}
```
Could it be an improvement?
In this way we avoid 2 optional chaining `?.` and 2 logical OR `||`
IMHO using several times the optional chaining could decrease code performance
Schermata 2021-08-04 alle 11 33 16

File: ts/components/screens/BaseScreenComponent.tsx [hunk 2]
```diff
@@ -21,24 +21,67 @@ import {
...
+          JSON.stringify(maybeSupportToken),
+          TypeLogs.INFO,
+          "support-token"
+        );
+      }
+      // set or remove the support token as instabug user attribute
+      setInstabugSupportTokenAttribute(maybeSupportToken);
+
+      if (payload.deviceUniqueId) {
+        setInstabugUserAttribute("deviceUniqueID", payload.deviceUniqueId);
```
Comment (Undermaken) [inline ts/components/screens/BaseScreenComponent.tsx:68]: I guess (actually I'm pretty sure) if a user sent its device ID, it persists also on future assistance requests
Could we remove this attribute if the user doesn't want to share it?
something like done in `setInstabugSupportTokenAttribute`

File: ts/components/screens/BaseScreenComponent.tsx [hunk 3]
```diff
@@ -324,3 +306,5 @@ export default connectStyle(
   {},
   mapPropsToStyleNames
 )(BaseScreenComponent);
+
+export { handleOnContextualHelpDismissed as test_handleOnContextualHelpDismissed };
```
Comment (Undermaken) [inline ts/components/screens/BaseScreenComponent.tsx:310]: If it is only for test purpose, could this solutions fit your needs?
https://github.com/pagopa/io-app/blob/51e0ff0ed4789304e2944b94a8213f63616c08fd/ts/components/wallet/PayWebViewModal.tsx#L209

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/components/screens/BaseScreenComponent.tsx (hunk 3 above)
Comment (pietro909) [reply ts/components/screens/BaseScreenComponent.tsx]: That's brilliant, thank you! Updated [here](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-99b6f65a74838b4555bfed0547c21e3f554b6fbca42f36eb9687795a8c2e96b6R308)

File: ts/components/ContextualHelp/CheckboxFormEntry.tsx (hunk 1 above)
Comment (pietro909) [reply ts/components/ContextualHelp/CheckboxFormEntry.tsx]: Very good point! I was copy-pasting blindly and missed this low-hanging fruit.
Fixed [as suggested](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-a3d54657eb55645a2e5cebccb5308f137cfaa6d26502e788ba303aac28c260daR82)

File: ts/components/screens/BaseScreenComponent.tsx (hunk 2 above)
Comment (pietro909) [reply ts/components/screens/BaseScreenComponent.tsx]: Good catch.
I've changed [this line](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-99b6f65a74838b4555bfed0547c21e3f554b6fbca42f36eb9687795a8c2e96b6R67) and added an ad-hoc [test](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-88d9976dbc65c76157a0ed848973de00891123ff8787772003de32a36ff0d508R101).

While doing it, I've polished a bit [the Instabug helpers](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-e8e418fa64073850d308f5d207dcdde4fdc42dfe50a21276c96b8cce3c12a4c1R142) and added [missing tests](ts/boot/__tests__/configureInstabug.test.ts).

**Review by Undermaken — APPROVED** [review]

**Pull request merged**